Quantum-Control-Applications/Superconducting/Multi Flux Tunable transmon/Use Case 1 - Two qubit gate optimization with cryoscope/readme.md

```diff
@@ -0,0 +1,192 @@
+
+## 3. The QUA program and data processing
+The QUA program consists of three distinct parts:
+1. Measuring the flux line step response by implementing cryoscope.
+2. From this step response, derive the IIR and FIR filters to compensate for the distorted flux pulse.
+3. Characterize the corrections by implementing a two qubit SWAP spectroscopy from which the CZ gate can be calibrated.
+
+### 3.1 [Cryoscope](cryoscope.py)
+
+ The idea of the method is to measure the step response of the flux line using the qubit phase measured in a Ramsey-like experiment [[1]](#1). 
+ The possibly distorted response can then be pre-corrected by designing suitable IIR and FIR filters.
+
+This experiment and implementation in QUA has already been documented in the following [use-case](#2).
+The main difference here is that a constant amplitude flux pulse has been used to measure the step response of the flux line.
+
+
+### 3.2 [IIR and FIR filter implementation](cryoscope.py)
+
+The method used to design the filters is to fit the flux step response with one or several exponential decay functions 
+and use analytical formulas to derive the corresponding FIR and IIR taps. 
+This exponential behaviour is characteristic of the effect of a bias-T on the flux line.
+
+These functions can be found in [filter_functions.py](filter_functions.py) and their usage is shown in the snippet below.
+
+```python
+import scipy.optimize
+from filter_functions import expdecay, filter_calc
+## Fit step response with exponential
+[A_exp, tau_exp], _ = scipy.optimize.curve_fit(expdecay, xplot, step_response)
+print(f"A: {A_exp}\ntau: {tau_exp}")
+
+## Derive IIR and FIR corrections
+fir, iir = filter_calc(exponential=[(A, tau)])
+print(f"FIR: {fir}\nIIR: {iir}")
+```
+Once the filter parameters have been found, the configuration needs to be updated accordingly.
+An example is shown on the snippet below 
+and more details about the implementation of FIR and IIR filters in QUA can be found [here](https://qm-docs.qualang.io/guides/output_filter#id1).
+```python
+"controllers": {
+    "con1": {
+        "analog_outputs": {
+            1: {"offset": -0.0102 * 1}, # q0 I
+            2: {"offset": 0.0316 * 1},  # q0 Q
+            3: {"offset": 0.0},  # q1 I
+            4: {"offset": 0.0},  # q1 Q
+            5: {"offset": 0.0},  # resonators I
+            6: {"offset": 0.0},  # resonators Q
+            7: {"offset": offset7, "filter": {"feedforward": fir, "feedback": iir}},   # qo flux line
+            8: {"offset": offset8, "filter": {"feedforward": fir1, "feedback": iir1}}, # q1 flux line
+        },
+    },
+},
+```
+The figure below shows the results of the filter implementation.
+
+The top-left figure represents the step response of the flux line and the crorresponding exponential fit.
+The top-right figure represents the filters correction as well as the calculated taps.

```

Comment:
The top right is confusing because it contains both experimental data and the scipy data. It is not clear that the red curve is scipy. I would suggest one of the two:
1. Remove the experimental data and indicate that this is all calculations
2. OR: "The top-right figure shows the experimental data in blue, the fitted curve in orange, the ideal pulse in XXX, and the calculated response with the filters in red"

Reply:
I lean toward the second option